src/hooks/useChannel.ts

```diff
@@ -23,24 +23,20 @@ export function useChannel(
     channelNameOrNameAndOptions: ChannelParameters,
     ...channelSubscriptionArguments: any[]
 ): ChannelAndClient {
-    const ably = useAbly(
+    const channelOptions =
         typeof channelNameOrNameAndOptions === 'object'
-            ? channelNameOrNameAndOptions.id
-            : undefined
-    );
-
-    const channelName =
-        typeof channelNameOrNameAndOptions === 'string'
             ? channelNameOrNameAndOptions
-            : channelNameOrNameAndOptions.channelName;
+            : { channelName: channelNameOrNameAndOptions };
+
+    const ably = useAbly(channelOptions.id);
 
-    const channel =
-        typeof channelNameOrNameAndOptions === 'string'
-            ? ably.channels.get(channelName)
-            : ably.channels.get(
-                  channelName,
-                  channelNameOrNameAndOptions.options
-              );
+    const { channelName } = channelOptions;
+
+    const channel = useMemo(
+        () => ably.channels.get(channelName, channelOptions.options),
+        // ignoring options change
+        [channelName]
```

Comment:
I realise I didn't capture this in the issue description but it might be worth putting the options in a ref and using `useEffect` to call `channel.setOptions` if they change?

Reply:
It's great idea!